File: dispute-portal-ui/src/modules/transactions/models/transaction.model.ts
export interface DisputeSummary {
  id: string
  referenceNumber: string
  status: string
  submittedAt: string
}

export interface AccountInfo {
  id: string
  accountName: string
  accountType: string
  accountNumber: string
}

export interface TransactionListItem {
  id: string
  transactionDate: string
  description: string
  amount: number
  category: string
  reference: string
  isDisputed: boolean
  disputeReference?: string
  accountId: string
  accountName: string
  accountType: string
  accountNumber: string
}

export interface TransactionDetail extends TransactionListItem {
  customerId: string
  createdAt: string
  dispute?: DisputeSummary
  account: AccountInfo
}

File: DisputePortalApi/DisputePortal.Application/Domain/Models/TransactionModel.cs
using DisputePortal.Application.Domain.Enums;

namespace DisputePortal.Application.Domain.Models;

public class TransactionModel
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid AccountId { get; set; }
    public decimal Amount { get; set; }
    public string Description { get; set; } = string.Empty;
    public TransactionCategory Category { get; set; }
    public DateTime TransactionDate { get; set; }
    public string Reference { get; set; } = string.Empty;
    public bool IsDisputed { get; set; }
    public DateTime CreatedAt { get; set; }

    public CustomerModel Customer { get; set; } = null!;
    public BankAccountModel Account { get; set; } = null!;
    public DisputeModel? Dispute { get; set; }
}
